### PR TITLE
OME teams: add schema.org markup

### PR DESCRIPTION
**Pull request by sbesson**
This PR explores options to improve the structure of the teams page using the schema.org markup.

The YAML front matter is converted into the equivalent schema.org representation (using a combination of https://schema.org/Organization and https://schema.org/Person). Some markup is also included to the standard `head.html` to markup the structured data as JSON-LD and make it available via `<script>`

Three options are investigated:

- 303582a keeps the structured metadata as [front matter](https://jekyllrb.com/docs/front-matter/) but includes the relevant schema.org markup
- 370bd4c migrates the structured metadata as a JSON [data file](https://jekyllrb.com/docs/datafiles/). The new `schemadotorg` page variable now becomes of pointer to the relevant data file
-  66daf3b is an alternative to 370bd4c where the structure metadata is stored in YAML format

### CODE REVIEW

**Comment by joshmoore** [comment]
Screen Shot 2021-03-29 at 11 13 52

Looks like something is wrong in the header syntax.

**Comment by sbesson** [comment]
> Looks like something is wrong in the header syntax.

Ha thanks, I got tripped by local caching using my browser `view-source`. The last commit should try to unwrap the markup from the YAML file and convert it into JSON. Locally `_site/teams/index.html ` now contains

```
        <script type="application/ld+json">
          
          {"@context":"https://schema.org","@type":"Organization","member":[{"@type":"Person","givenName":"Jason","familyName":"Swedlow","jobTitle":"Co-Founder","description":"Jason Swedlow earned a BA in Chemistry from Brandeis University in 1982. He performed his PhD in Biophysics with Profs D. A. Agard and J. W. Sedat, finishing in 1994. Dr Swedlow was a postdoctoral fellow at UCSF and then Harvard Medical School from 1994 and 1998, supported by a Damon Runyon Walter Winchell Cancer Research Fund Fellowship from 1995 to 1997. In 1998, Dr Swedlow established his own laboratory at the Wellcome Trust Biocentre, University of Dundee, Scotland as a Principal Investigator and Wellcome Trust Career Development Fellow. He was awarded a Wellcome Trust Senior Fellowship in 2002...
```

**Comment by joshmoore** [comment]
:+1: https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fsnoopycrimecop.github.io%2Fwww.openmicroscopy.org%2Fteams%2F (deprecated) gives no errors or warnings.

File: _data/swedlowlab.json [hunk 1]
```diff
@@ -0,0 +1,216 @@
+{
+  "@context": "https://schema.org",
+  "@type": "Organization",
+  "member": [
+    {
+      "@type": "Person",
+      "givenName": "Jason",
+      "familyName": "Swedlow",
+      "jobTitle": "Co-Founder",
+      "description": "Jason Swedlow earned a BA in Chemistry from Brandeis University in 1982. He performed his PhD in Biophysics with Profs D. A. Agard and J. W. Sedat, finishing in 1994. Dr Swedlow was a postdoctoral fellow at UCSF and then Harvard Medical School from 1994 and 1998, supported by a Damon Runyon Walter Winchell Cancer Research Fund Fellowship from 1995 to 1997. In 1998, Dr Swedlow established his own laboratory at the Wellcome Trust Biocentre, University of Dundee, Scotland as a Principal Investigator and Wellcome Trust Career Development Fellow. He was awarded a Wellcome Trust Senior Fellowship in 2002, and named Professor of Quantitative Cell Biology in 2007. His lab focuses on studies of mitotic and interphase chromosome structure and dynamics. He is Co-Director of the Analytical and Quantitative Microscopy Course. He is co-founder of OME (along with Peter Sorger and Ilya Goldberg)."
```
Comment (joshmoore) [inline _data/swedlowlab.json:10]: Line length is probably the major downside of the JSON strategy,no ?

File: _includes/head.html [hunk 2]
```diff
@@ -13,6 +13,12 @@
   gtag('config', '{{ site.ga_token }}');
 
         </script>
+        {% if page.schemadotorg %}
+        <script type="application/ld+json">
+          {% assign var = page.schemadotorg %}
+          {{ site.data[var] | jsonify}}
```
Comment (joshmoore) [inline _includes/head.html:19]: This is pretty super long when inspecting the source. Perhaps pipe to `pretty_json` as well?

**Review by joshmoore — APPROVED** [review]
Generally happy to move forward with this. Is there any way to access the json files directly or do they disappear in jekyll?

File: _data/swedlowlab.json (hunk 1 above)
Comment (sbesson) [reply _data/swedlowlab.json:10]: Definitely, I tried investigating a bit into https://gdevops.gitlab.io/tuto_git/tools/pre-commit/repos_hooks/repo_pre_commit_hooks/pretty-format-json/pretty-format-json.html but I could not find a built-in option to set a max-line-length and break a long string into an array representation.

**Comment by sbesson** [comment]
> Is there any way to access the json files directly or do they disappear in jekyll?

I agree that `JSON` -> Hash -> `JSON` feels unnecessary. My searches for trying access the raw content of the data files  have been unsuccessful so far.

File: _includes/head.html (hunk 2 above)
Comment (sbesson) [reply _includes/head.html:19]: After a bit of digging there does not seem to be any built-in `pretty_json` filter. Possible something to capture as an extension? Or rely on tooling like https://search.google.com/structured-data/testing-tool/ to expose the JSON in a more parsable way?

**Pull request merged**